### ISSUE TITLE
Support for using string data in alerts both for expressions and annotations

### ISSUE THREAD

**Issue by ChristianHoumann** [grafana/grafana#88616]
**Why is this needed**:

This feature would be useful in order to know more specific things about devices that trigger an alert. Say that one out of many devices in a network reaches a specific fail state that you want to alert on, right now you can only know that a device has reached that state, not its name or other identifiers. It would make alerting much more powerful to be able to include string info so that the ones responding to the alert know excactly which device has failed. I imagine an alert where one query is the alert condition which checks if a device has reached a certain fail state, and another query get the name and other info about the device for use in the alert text.

**What would you like to be added**:

I would like this feature to support querying for string data, and making alert conditions through expressions that match the string in different ways, such as regex. I also most importantly this feature to make it possible to be able to take string data from a query and be able to input this in the alert summary or annotations, or even title in a way like "{A.device_id} has failed". 

Basically i want an alert to be able to say "Device xxxxxx has reached a failstate" instead of "A device has reached a failstate".

**Who is this feature for?**

This feature is for users who work not just with stream data, but where it is also important to know states about specific IoT devices or servers within a large network.

Im sorry if this has already been requested, i think it could be a really powerful feature. I hope its clear what i mean, and please let me know if this is possible in any way already.
